invenio/modules/knowledge/api.py

```diff
@@ -547,32 +547,34 @@ def get_kbd_values_by_def(confdict, searchwith=""):
     collection = ""
     if 'collection' in confdict:
         collection = confdict['collection']
-    reclist = []  # return this
+    response = None
     if searchwith and expression:
         if (expression.count('%') > 0):
             expression = expression.replace("%", searchwith)
-            reclist = search_engine.perform_request_search(p=expression,
-                                                           cc=collection)
+            response = Query(expression).search(collection=collection)
         else:
             # no %.. just make a combination
             expression = expression + " and " + searchwith
-            reclist = search_engine.perform_request_search(p=expression,
-                                                           cc=collection)
+            response = Query(expression).search(collection=collection)
     else:  # either no expr or no searchwith.. but never mind about searchwith
         if expression:  # in this case: only expression
-            reclist = search_engine.perform_request_search(p=expression,
-                                                           cc=collection)
+            response = Query(expression).search(collection=collection)
         else:
             # make a fake expression so that only records that have this field
             # will be returned
             fake_exp = "/.*/"
             if searchwith:
                 fake_exp = searchwith
-            reclist = search_engine.perform_request_search(f=field, p=fake_exp,
-                                                           cc=collection)
-    if reclist:
-        return [val for (val, dummy) in
-                search_engine.get_most_popular_field_values(reclist, field)]
+            response = Query("{0}:{1}".format(field, fake_exp)).search(
+                collection=collection
+            )
+    if response.recids:
```

Comment:
@jalavik can we just use `response.records()` as usage of `response.recids` is very heavy? We might want to return only `field` in search results and also we need to set correct pagination.